components/pthread/include/esp_pthread.h

```diff
@@ -24,11 +27,33 @@ extern "C" {
 
 /** pthread configuration structure that influences pthread creation */
 typedef struct {
-    size_t stack_size;    ///< the stack size of the pthread
-    size_t prio;          ///< the thread's priority
-    bool inherit_cfg;     ///< inherit this configuration further
+    size_t stack_size;  ///< The stack size of the pthread
+    size_t prio;        ///< The thread's priority
+    bool inherit_cfg;   ///< Inherit this configuration further
+    // This member has to be an array and not a pointer due to the fact that the entire
+    // struct is copied by the thread when inherit_cfg is true which in turn likely
+    // will result in dangling pointers, especially if threads are created/destroyed
+    // dynamically at run time with different names etc.
+    char thread_name[configMAX_TASK_NAME_LEN + 1];  ///< The thread name.
+    int pin_to_core;    ///< The core id to pin the thread to. See components/pthread/KConfig for valid ranges.
 } esp_pthread_cfg_t;
 
+/**
+ * @brief Creates a default pthread configuration based
+ * on the values set via menuconfig.
+ * 
+ * @return
+ *      A default configuration structure.
+ */
+esp_pthread_cfg_t esp_create_default_pthread_config();
```

Comment:
Can you make it esp_pthread_get_default_config (so that API name is prefixed with esp_pthread)?

Same with esp_pthread_set_name below.

Reply:
Yes.